### PR TITLE
rpc: fix crash 'listforwards' when payment_hash is empty

### PR DESCRIPTION
**Pull request by niftynei**
There's a forward on one of my node's that's missing a payment_hash, which caused a crash when calling `listforwards`. This patch fixes the crash, not sure what the root cause is.

```
2019-08-14T17:50:39.100Z **BROKEN** lightningd(11355): FATAL SIGNAL 11 (version v0.7.2rc1)
2019-08-14T17:50:39.100Z **BROKEN** lightningd(11355): backtrace: common/daemon.c:45 (send_backtrace) 0x563349d07879
2019-08-14T17:50:39.100Z **BROKEN** lightningd(11355): backtrace: common/daemon.c:53 (crashdump) 0x563349d078c9
2019-08-14T17:50:39.100Z **BROKEN** lightningd(11355): backtrace: (null):0 ((null)) 0x7efd7b996f1f
2019-08-14T17:50:39.100Z **BROKEN** lightningd(11355): backtrace: ccan/ccan/str/hex/hex.c:59 (hex_encode) 0x563349d57fec
2019-08-14T17:50:39.100Z **BROKEN** lightningd(11355): backtrace: lightningd/json.c:380 (json_add_hex) 0x563349cd9dd3
2019-08-14T17:50:39.100Z **BROKEN** lightningd(11355): backtrace: lightningd/peer_htlcs.c:2151 (json_format_forwarding_object) 0x563349cfa7ac
2019-08-14T17:50:39.100Z **BROKEN** lightningd(11355): backtrace: lightningd/peer_htlcs.c:2198 (listforwardings_add_forwardings) 0x563349cfa99d
2019-08-14T17:50:39.100Z **BROKEN** lightningd(11355): backtrace: lightningd/peer_htlcs.c:2216 (json_listforwards) 0x563349cfaa55
2019-08-14T17:50:39.100Z **BROKEN** lightningd(11355): backtrace: lightningd/jsonrpc.c:650 (parse_request) 0x563349cdc184
2019-08-14T17:50:39.100Z **BROKEN** lightningd(11355): backtrace: lightningd/jsonrpc.c:748 (read_json) 0x563349cdc5ae
2019-08-14T17:50:39.100Z **BROKEN** lightningd(11355): backtrace: ccan/ccan/io/io.c:59 (next_plan) 0x563349d4bbe5
2019-08-14T17:50:39.100Z **BROKEN** lightningd(11355): backtrace: ccan/ccan/io/io.c:407 (do_plan) 0x563349d4c762
2019-08-14T17:50:39.101Z **BROKEN** lightningd(11355): backtrace: ccan/ccan/io/io.c:417 (io_ready) 0x563349d4c7a0
2019-08-14T17:50:39.101Z **BROKEN** lightningd(11355): backtrace: ccan/ccan/io/poll.c:445 (io_loop) 0x563349d4e7f5
2019-08-14T17:50:39.101Z **BROKEN** lightningd(11355): backtrace: lightningd/io_loop_with_timers.c:24 (io_loop_with_timers) 0x563349cd8afe
2019-08-14T17:50:39.101Z **BROKEN** lightningd(11355): backtrace: lightningd/lightningd.c:834 (main) 0x563349cded3a
2019-08-14T17:50:39.101Z **BROKEN** lightningd(11355): backtrace: (null):0 ((null)) 0x7efd7b979b96
2019-08-14T17:50:39.101Z **BROKEN** lightningd(11355): backtrace: (null):0 ((null)) 0x563349cc5909
2019-08-14T17:50:39.101Z **BROKEN** lightningd(11355): backtrace: (null):0 ((null)) 0xffffffffffffffff
```

### CODE REVIEW

File: lightningd/peer_htlcs.c [hunk 1]
```diff
@@ -2148,9 +2148,12 @@ void json_format_forwarding_object(struct json_stream *response,
...
 
-	json_add_hex(response, "payment_hash",
-				    cur->payment_hash,
-				    sizeof(*cur->payment_hash));
+	if (cur->payment_hash)
+		json_add_hex(response, "payment_hash",
+					    cur->payment_hash,
+					    sizeof(*cur->payment_hash));
+	else
+		json_add_string(response, "payment_hash", "UNKNOWN");
```
Comment (ZmnSCPxj) [inline lightningd/peer_htlcs.c]: Would it be better to give this as `"payment_hash" : null`? Or would that make trouble for most JSON libraries in other languages?

Comment (rustyrussell) [reply lightningd/peer_htlcs.c]: Neither.  If you don't know it, you should omit the field.

**Comment by rustyrussell** [comment]
6d333f16cc0f3aac7097269bf0985b5fa06d59b4 explains it.  We can delete the HTLC details, apparently.  So we need to omit this field if NULL.

File: lightningd/peer_htlcs.c (hunk 1 above)
Comment (niftynei) [reply lightningd/peer_htlcs.c]: ack

**Comment by niftynei** [comment]
ACK https://github.com/ElementsProject/lightning/pull/2954/commits/73f1605e2873867ed803ff75045cc9fccf2ca0a1

**Pull request merged**